recipes-kernel/linux/files/patches-6.1/0083-dts-iot2050-Support-IOT2050-SM-variant.patch

```diff
@@ -0,0 +1,383 @@
++	/delete-property/ pinctrl-27;
++	/delete-property/ pinctrl-28;
++	/delete-property/ pinctrl-29;
++	/delete-property/ pinctrl-30;
++	/delete-property/ pinctrl-31;
++	/delete-property/ pinctrl-32;
++	/delete-property/ pinctrl-33;
++	/delete-property/ pinctrl-34;
++	/delete-property/ pinctrl-35;
++	/delete-property/ pinctrl-36;
++	/delete-property/ pinctrl-37;
++	/delete-property/ pinctrl-38;
++	/delete-property/ pinctrl-39;
++	/delete-property/ pinctrl-40;
++	/delete-property/ pinctrl-41;
++	/delete-property/ pinctrl-42;
++	/delete-property/ pinctrl-43;
++	/delete-property/ pinctrl-44;
++	/delete-property/ pinctrl-45;
++	/delete-property/ pinctrl-46;
++	/delete-property/ pinctrl-47;
++	/delete-property/ pinctrl-48;
++	/delete-property/ pinctrl-49;
++	/delete-property/ pinctrl-50;
++
++	user1_led_pins: user1-led-default-pins {
++		pinctrl-single,pins = <
++			/* (AB1) WKUP_UART0_RXD:WKUP_GPIO0_52, as USER 1 led red */
++			AM65X_WKUP_IOPAD(0x00a0, PIN_OUTPUT, 7)
++			/* (AB5) WKUP_UART0_TXD:WKUP_GPIO0_53, as USER 1 led green */
++			AM65X_WKUP_IOPAD(0x00a4, PIN_OUTPUT, 7)
++		>;
++	};
++
++	soc_asic_pins: soc-asic-default-pins {
++		pinctrl-single,pins = <
++			AM65X_WKUP_IOPAD(0x0044, PIN_INPUT, 7)  /* (P4) WKUP_GPIO0_29 */
++			AM65X_WKUP_IOPAD(0x0048, PIN_INPUT, 7)  /* (P5) WKUP_GPIO0_30 */
++			AM65X_WKUP_IOPAD(0x004c, PIN_INPUT, 7)  /* (P1) WKUP_GPIO0_31 */
++		>;
++	};
++};
++
++&main_gpio0 {
++	gpio-line-names = "main_gpio0-base";
++};
++
++&main_gpio1 {
++	pinctrl-names = "default";
++	pinctrl-0 =
++		<&cp2102n_reset_pin_default>,
++		<&main_pcie_enable_pins_default>,
++		<&asic_spi_mux_ctrl_pin>;
++	gpio-line-names =
++		/* 0..9 */
++		"", "", "", "", "", "", "", "", "", "",
++		/* 10..19 */
++		"", "", "", "", "", "", "", "", "", "",
++		/* 20..29 */
++		"", "", "", "", "CP2102N-RESET", "", "", "", "", "",
++		/* 30..39 */
++		"", "", "", "", "", "", "", "", "", "",
++		/* 40..49 */
++		"", "", "", "", "", "", "", "", "", "",
++		/* 50..59 */
++		"", "", "", "", "", "", "", "", "", "",
++		/* 60..69 */
++		"", "", "", "", "", "", "", "", "", "",
++		/* 70..79 */
++		"", "", "", "", "", "", "", "", "", "",
++		/* 80..86 */
++		"", "", "", "", "", "", "ASIC-spi-mux-ctrl";
++};
++
++&wkup_gpio0 {
++	pinctrl-names = "default";
++	pinctrl-0 =
++		<&push_button_pins_default>,
++		<&db9_com_mode_pins_default>,
++		<&soc_asic_pins>;
++	gpio-line-names =
++		/* 0..9 */
++		"wkup_gpio0-base", "", "", "", "UART0-mode1", "UART0-mode0",
++		"UART0-enable", "UART0-terminate", "", "WIFI-disable",
++		/* 10..19 */
++		"", "", "", "", "", "", "", "", "", "",
++		/* 20..29 */
++		"", "", "", "", "", "USER-button", "", "", "","ASIC-gpio-0",
++		/* 30..31 */
++		"ASIC-gpio-1", "ASIC-gpio-2";
++};
++
++&main_spi0 {
++	pinctrl-names = "default";
++	pinctrl-0 = <&main_spi0_pins>;
++
++	#address-cells = <1>;
++	#size-cells= <0>;
++
++	spidev@0 {
++		compatible = "rohm,dh2228fv";
++		spi-max-frequency = <20000000>;
++		reg = <0>;
++	};
```

Comment:
Split the spidev node off into a separate commit - it can't be part of mainline. We still need a different solution for that problem. I'm thinking about enhancing SPI controllers by a why to instantiate a new device node during runtime via sysfs, allowing to bind spidev to that dynamically. But not code for that yet, and no idea if that will be accepted upstream.